tests/foreman/cli/test_leapp_client.py

```diff
@@ -0,0 +1,262 @@
+"""Test for LEAPP cli
+
+:Requirement: leapp
+
+:CaseLevel: Acceptance
+
+:CaseComponent: Leappintegration
+
+:Team: Rocket
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:CaseAutomation: Automated
+
+:Upstream: No
+"""
+import pytest
+from wait_for import wait_for
+
+from robottelo.constants import PRDS
+
+CUSTOM_REPOSET = {
+    'rhel8_8_bos': 'Red Hat Enterprise Linux 8 for x86_64 - BaseOS (RPMs)',
+    'rhel8_8_aps': 'Red Hat Enterprise Linux 8 for x86_64 - AppStream (RPMs)',
+    'rhel9_2_bos': 'Red Hat Enterprise Linux 9 for x86_64 - BaseOS (RPMs)',
+    'rhel9_2_aps': 'Red Hat Enterprise Linux 9 for x86_64 - AppStream (RPMs)',
+}
+CUSTOM_REPOS = {
+    'rhel8_8_bos': {
+        'id': 'rhel-8-for-x86_64-baseos-rpms',
+        'name': 'Red Hat Enterprise Linux 8 for x86_64 - BaseOS RPMs 8.8',
+        'releasever': '8.8',
+    },
+    'rhel8_8_aps': {
+        'id': 'rhel-8-for-x86_64-appstream-rpms',
+        'name': 'Red Hat Enterprise Linux 8 for x86_64 - AppStream RPMs 8.8',
+        'releasever': '8.8',
+    },
+    'rhel9_2_bos': {
+        'id': 'rhel-9-for-x86_64-baseos-rpms',
+        'name': 'Red Hat Enterprise Linux 9 for x86_64 - BaseOS RPMs 9.2',
+        'releasever': '9.2',
+    },
+    'rhel9_2_aps': {
+        'id': 'rhel-9-for-x86_64-appstream-rpms',
+        'name': 'Red Hat Enterprise Linux 9 for x86_64 - AppStream RPMs 9.2',
+        'releasever': '9.2',
+    },
+}
+
+
+def create_activation_key(satellite, content_view, lifecycle_env, organization):
+    """Create activation key uinsg specific entities"""
+    return satellite.api.ActivationKey(
+        content_view=content_view,
+        environment=lifecycle_env,
+        organization=organization,
+    ).create()
+
+
+def register_host_with_satellite(satellite, custom_host, organization, activation_key):
+    """Register content host with satellite"""
+    custom_host.install_katello_ca(satellite)
+    custom_host.register_contenthost(org=organization.label, activation_key=activation_key.name)
```

Comment:
- Use `custom_host.register` instead of `custom_host.register_contenthost` 
- remove `custom_host.install_katello_ca` and `custom_host.add_rex_key`

Reply:
I tried the way suggested, but unfortunately getting an error

> stderr:
    (213, b'Error: There are no enabled repositories in "/etc/yum.repos.d", "/etc/yum/repos.d", "/etc/distro.repos.d".\nCouldn\'t find activation key \'robottelo.hosts.DecClass(auto_attach=True\' (HTTP error code 404: Not Found)\n')
    status: 1

I think we can take this change in upcoming PR's, will that be ok @Gauravtalreja1